src/types.rs

```diff
@@ -139,15 +142,27 @@ impl Request {
             method: req.method().clone(),
             params: HashMap::new(),
             body,
+            path: req.path().to_string(),
+            host: req.connection_info().host().to_string(),
+            scheme: req.connection_info().scheme().to_string(),
         }
     }
 
     pub fn to_hashmap(&self, py: Python<'_>) -> Result<HashMap<&str, Py<PyAny>>> {
         let mut result = HashMap::new();
+
+        result.insert("method", self.method.as_ref().to_object(py));
         result.insert("params", self.params.to_object(py));
         result.insert("queries", self.queries.to_object(py));
         result.insert("headers", self.headers.to_object(py));
         result.insert("body", self.body.to_object(py));
+
+        let mut url = HashMap::new();
+        url.insert("path", self.path.to_string());
+        url.insert("host", self.host.to_string());
+        url.insert("scheme", self.scheme.to_string());
+        result.insert("url", url.to_object(py));
+
```

Comment:
@r3b-fish , all looks good to me. But what is the need to create a new hasmap here? Why not return all the params as is?

Reply:
FastAPI uses [the same structure](https://www.starlette.io/requests/#url). I think we may implement it from a popular solution.

Reply:
I may rewrite if you want, no problem, it doesn't matter for me

Reply:
No, exposing a URL object looks like a very nice interface. Thank you for this idea :smile: